src/lib/optimizer/strategy/stored_table_column_alignment_rule.hpp

```diff
@@ -0,0 +1,54 @@
+#pragma once
+
+#include <unordered_map>
+#include <vector>
+
+#include "abstract_rule.hpp"
+#include "logical_query_plan/stored_table_node.hpp"
+
+namespace opossum {
+
+// Modified hash code generation for StoredTableNodes where column pruning information is omitted. Struct is used to
+// enable hash based containers containing std::shared_ptr<StoredTableNode>.
+struct StoredTableNodeSharedPtrHash final {
```

Comment:
These strucs, as well as StoredTableNodeUnorderedMap, are not used outside the rule, are they? Move into an anonymous namespace within the cpp?

Also, `StoredTableNodeUnorderedMap` sounds a bit incorrect. We have specific equality semantics here and the name should cover that. ColumnPruningAgnosticSomething?